common/trtexec_vendor/CMakeLists.txt

```
@@ -0,0 +1,73 @@
+cmake_minimum_required(VERSION 3.14)
+project(trtexec_vendor)
+
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(CUDA REQUIRED)
+find_package(cudnn_cmake_module REQUIRED)
+find_package(CUDNN REQUIRED)
+find_package(tensorrt_cmake_module REQUIRED)
+find_package(TENSORRT REQUIRED)
+
+if(NOT (${CUDA_FOUND} AND ${CUDNN_FOUND} AND ${TENSORRT_FOUND}))
+  message(WARNING "cuda, cudnn, tensorrt libraries are not found")
+  return()
+endif()
+
+set(TRTEXEC_DEFAULT_BIN /usr/src/tensorrt/bin/trtexec)
+if(NOT EXISTS TRTEXEC_DEFAULT_BIN)
+  include(FetchContent)
+  if(${TENSORRT_VERSION} VERSION_EQUAL 8.4.2)
+    set(TENSORRT_VERSION 8.4.1)
+  endif()
```

```suggestion
  if(${TENSORRT_VERSION} VERSION_EQUAL 8.4.2)
    set(TENSORRT_VERSION 8.4.1)
  endif()
  if(${TENSORRT_VERSION} VERSION_EQUAL 8.2.4)
    set(TENSORRT_VERSION 8.2.1)
  endif()
```
I think we may still support the previous CUDA environemt. (https://github.com/autowarefoundation/autoware/pull/2766)